### PR TITLE
Display Entry points information & Add Sort feature

### PR DESCRIPTION
**Pull request by AhmedBasem20**
In this PR, the JSON file is modified to include all the entry points information. and then display it in the `EntryPoints` React component.
I also added a sort feature using two options, alphabetical and commits count. These options can be selected using a dropdown list.

## Issues
 * The `webpage` is slow (~35 min) because it depends on `test-install` workflow where the entry points information is collected. @ltalirz introduced a solution here https://github.com/aiidateam/aiida-registry/pull/240#discussion_r1241079352 but I'm not sure when we need to run the `test-install` and when we don't.
 * The description of the inputs/outputs is usually incomplete. 
image
Not sure if it is an issue with data collection or there is no issue to display it as it is.

### CODE REVIEW

**Comment by unkcpz** [comment]
@AhmedBasem20 thanks a lot, this is fantastic! I'll have a close look after our meeting.

> Not sure if it is an issue with data collection or there is no issue to display it as it is.

I think it is better to show all descriptions and wrap the words properly. 

> but I'm not sure when we need to run the test-install and when we don't.

Let's discuss it after.

**Comment by AhmedBasem20** [comment]
Hi @unkcpz, the incomplete description is fixed. As you said, it was not an issue with `aiida-core`.
I think this PR is ready for review now. The new information is displayed in a table-based structure, I'm not sure if it is the best way to display it, if you have any suggestions please let me know.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -52,7 +52,7 @@ jobs:
     - name: Install npm dependencies and build
       run: |
         npm install
-        export BASE_PATH="/aiida-registry/pr-preview/pr-${{ github.event.number }}/"
+        export VITE_BASE_PATH="/aiida-registry/pr-preview/pr-${{ github.event.number }}/"
```
Comment (unkcpz) [inline .github/workflows/ci.yml:55]: The environment variable is for pr-preview only? What is after the pr merged to the main, does this path needs to be changed?

File: aiida-registry-app/src/App.jsx [hunk 2]
```diff
@@ -8,15 +8,20 @@ import MaX from './assets/MaX.png'
...
+import Select from '@mui/material/Select';
 
 
 const entrypointtypes = jsonData["entrypointtypes"]
 const globalsummary = jsonData["globalsummary"]
 const plugins  = jsonData["plugins"]
 const status_dict = jsonData["status_dict"]
 const length = Object.keys(plugins).length;
-const currentPath = window.location.pathname;
+const currentPath = import.meta.env.VITE_BASE_PATH || "/aiida-registry/";
```
Comment (unkcpz) [inline aiida-registry-app/src/App.jsx:23]: Ah, I see, the `VITE_BASE_PATH` is used here. Can you add a comment on the CI action above about it usage here?

File: aiida-registry-app/src/App.jsx [hunk 3]
```diff
@@ -53,6 +58,31 @@ function App() {
...
+
+    let sortedPlugins;
+    if (option === 'commits') {
+      const pluginsArray = Object.entries(plugins);
+
+      // Sort the array based on the commit_count value
+      pluginsArray.sort(([, pluginA], [, pluginB]) => pluginB.commits_count - pluginA.commits_count);
+
+      // Create a new object with the sorted entries
+      sortedPlugins = Object.fromEntries(pluginsArray);
```
Comment (unkcpz) [inline aiida-registry-app/src/App.jsx:75]: Very nice, it is pretty clear, I am not very familiar with js and I can understand it in the first glimpse.

**Review by unkcpz — CHANGES_REQUESTED** [review]
@AhmedBasem20 thanks a lot! I just have a minor request.

Is that possible, in the description, the text quoted with "\`", such as "\`aiida\`" can be parsed with [code tag](https://www.w3schools.com/tags/tag_code.asp) and show things like "\`aiida\`" -> "`aiida`".

File: aiida-registry-app/src/App.jsx (hunk 2 above)
Comment (AhmedBasem20) [reply aiida-registry-app/src/App.jsx]: I noticed that it shouldn't be in the CI action at all. As it tests the webpage build, not the PR preview.
I renamed the environment variable to `VITE_PR_PREVIEW_PATH`. I think this is more descriptive.

**Comment by unkcpz** [comment]
Thanks @AhmedBasem20. It is all good for me. 

Although I don't think the checkbox is very pretty but it is there and we can find a beautiful one in the future. 
I'd propose this can be merged now and leave the issues to polish by another PR. This PR is introduced to bring the functionality.

Things to polish:
- [ ] The checkbox format and type.
- [ ] In the plugin details, add outlines of plugins and can redirect to the specific item.
- [ ] Each plugin name in the description page is better to be a bit larger.

I'll open an issue for the layout and UI polishment. 

I just approve of it. Feel free to have another look at this? @ltalirz

**Review by unkcpz — APPROVED** [review]

**Pull request merged**